cmd/promtool/main.go

```diff
@@ -353,8 +353,11 @@ func checkConfig(agentMode bool, filename string) ([]string, error) {
 	}
 
 	for _, scfg := range cfg.ScrapeConfigs {
-		if err := checkFileExists(scfg.HTTPClientConfig.BearerTokenFile); err != nil {
-			return nil, errors.Wrapf(err, "error checking bearer token file %q", scfg.HTTPClientConfig.BearerTokenFile)
+		authzSet := scfg.HTTPClientConfig.Authorization != nil
+		if authzSet {
```

Comment:
nit: this variable isn't really needed.

```suggestion
		if scfg.HTTPClientConfig.Authorization != nil {
```